clang/lib/Driver/Driver.cpp

```diff
@@ -925,8 +925,17 @@ void Driver::CreateOffloadingDeviceToolChains(Compilation &C,
 
           // Warn about deprecated `sycldevice` environment component.
           if (TT.getEnvironmentName() == "sycldevice") {
+            // Build a string with suggested target triple.
+            std::string SuggestedTriple = TT.getArchName().str();
+            if (TT.getOS() != llvm::Triple::UnknownOS) {
+              SuggestedTriple += '-';
+              if (TT.getVendor() != llvm::Triple::UnknownVendor)
+                SuggestedTriple += TT.getVendorName();
+              SuggestedTriple += Twine("-" + TT.getOSName()).str();
+            } else if (TT.getVendor() != llvm::Triple::UnknownVendor)
+              SuggestedTriple += Twine("-" + TT.getVendorName()).str();
```

Comment:
Would it make more sense to copy the triple, drop the environment and just print triple.str()?

Reply:
It would be easier to implement, but what I'm trying to do with all these jumping though the hoops is to improve the warning message. I wanted to promote usage of "short aliases" by suggesting `spir` instead of `spir-unknown-unknown` triple.
I'm okay to print full target triple if you think that code complications don't worth it.

Reply:
The code complications are centralized to the diagnostic - The promotion of the short alias would be preferred.